datafusion/core/tests/sql/window.rs

```diff
@@ -2129,16 +2131,17 @@ async fn test_window_agg_global_sort_intermediate_parallel_sort() -> Result<()>
     // Only 1 SortExec was added
     let expected = {
         vec![
-            "SortExec: expr=[c1@0 ASC NULLS LAST]",
-            "  ProjectionExec: expr=[c1@0 as c1, SUM(aggregate_test_100.c9) PARTITION BY [aggregate_test_100.c1] ORDER BY [aggregate_test_100.c9 ASC NULLS LAST] ROWS BETWEEN 1 PRECEDING AND 3 FOLLOWING@2 as sum1, SUM(aggregate_test_100.c9) ORDER BY [aggregate_test_100.c9 ASC NULLS LAST] ROWS BETWEEN 1 PRECEDING AND 5 FOLLOWING@3 as sum2]",
-            "    BoundedWindowAggExec: wdw=[SUM(aggregate_test_100.c9): Ok(Field { name: \"SUM(aggregate_test_100.c9)\", data_type: UInt64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }), frame: WindowFrame { units: Rows, start_bound: Preceding(UInt64(1)), end_bound: Following(UInt64(5)) }]",
-            "      SortPreservingMergeExec: [c9@1 ASC NULLS LAST]",
-            "        SortExec: expr=[c9@1 ASC NULLS LAST]",
-            "          BoundedWindowAggExec: wdw=[SUM(aggregate_test_100.c9): Ok(Field { name: \"SUM(aggregate_test_100.c9)\", data_type: UInt64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }), frame: WindowFrame { units: Rows, start_bound: Preceding(UInt64(1)), end_bound: Following(UInt64(3)) }]",
-            "            SortExec: expr=[c1@0 ASC NULLS LAST,c9@1 ASC NULLS LAST]",
-            "              CoalesceBatchesExec: target_batch_size=8192",
-            "                RepartitionExec: partitioning=Hash([Column { name: \"c1\", index: 0 }], 2), input_partitions=2",
-            "                  RepartitionExec: partitioning=RoundRobinBatch(2), input_partitions=1",
+            "SortPreservingMergeExec: [c1@0 ASC NULLS LAST]",
+            "  SortExec: expr=[c1@0 ASC NULLS LAST], global=false",
```

Comment:
It seems that final `SortPreservingMergeExec` in the plan is redundant. Its input has already single partition. What do you think?

Reply:
Agree with you. It is redundant.

Reply:
There is still some optimization room in the TopDown rule regarding the window expr ordering and global ordering.